File: node/src/services/riskService.js
export class RiskService {
  /**
   * Assess a renter's risk based on their profile.
   * @param {Object} profile - The renter's profile data
   * @returns {Object} The result of the risk assessment
   */
  assessRenter(profile) {
    const { annual_income, credit_score, monthly_rent } = profile;
    const rentToIncomeRatio = (monthly_rent * 12) / annual_income;

    let riskScore = 50; // Base score

    // Credit score impact
    if (credit_score > 750) {
      riskScore -= 30;
    } else if (credit_score > 650) {
      riskScore -= 15;
    } else if (credit_score < 600) {
      riskScore += 25;
    }

    // Income ratio impact
    if (rentToIncomeRatio > 0.4) {
      riskScore += 20;
    } else if (rentToIncomeRatio < 0.25) {
      riskScore -= 10;
    }

    // Qualification criteria
    const qualified = riskScore < 70 && credit_score >= 600;

    let recommendation = "Approve";
    if (!qualified) {
      recommendation = "Reject - High Risk";
    } else if (riskScore > 40) {
      recommendation = "Approve with Co-signer";
    }

    const maxCoverage = qualified ? monthly_rent * 12 : 0.0;

    return {
      renter_id: profile.id,
      risk_score: riskScore,
      qualified: qualified,
      max_coverage: maxCoverage,
      recommendation: recommendation,
      evaluated_at: new Date().toISOString()
    };
  }
}

File: ruby/app/services/risk_service.rb
require 'time'

class RiskService
  def assess_renter(profile)
    annual_income = profile['annual_income'].to_f
    credit_score = profile['credit_score'].to_i
    monthly_rent = profile['monthly_rent'].to_f

    rent_to_income_ratio = (monthly_rent * 12) / annual_income
    
    risk_score = 50 # Base score

    # Credit score impact
    if credit_score > 750
      risk_score -= 30
    elsif credit_score > 650
      risk_score -= 15
    elsif credit_score < 600
      risk_score += 25
    end

    # Income ratio impact
    if rent_to_income_ratio > 0.4
      risk_score += 20
    elsif rent_to_income_ratio < 0.25
      risk_score -= 10
    end

    # Qualification criteria
    qualified = risk_score < 70 && credit_score >= 600
    
    recommendation = "Approve"
    if !qualified
      recommendation = "Reject - High Risk"
    elsif risk_score > 40
      recommendation = "Approve with Co-signer"
    end

    max_coverage = qualified ? monthly_rent * 12 : 0.0

    {
      renter_id: profile['id'],
      risk_score: risk_score,
      qualified: qualified,
      max_coverage: max_coverage,
      recommendation: recommendation,
      evaluated_at: Time.now.iso8601
    }
  end
end
